src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/hotheap/HighHeapUsageYoungGenRca.java

```diff
@@ -118,6 +133,67 @@ protected void remove(SlidingWindowData e) {
     this(rcaPeriod, 1.0, heap_Used, gc_Collection_Time);
   }
 
+  private boolean fullGcTimeTooHigh(double avgFullGcTime) {
+    return (!Double.isNaN(avgFullGcTime) && avgFullGcTime > FOLLOWER_CHECK_TIMEOUT_MS);
+  }
+
+  private boolean promotionRateTooHigh(double avgPromotionRate, double modifier) {
+    return (!Double.isNaN(avgPromotionRate) && avgPromotionRate > promotionRateThreshold * modifier);
+  }
+
+  private boolean youngGcTimeTooHigh(double avgYoungGCTime) {
+    return (!Double.isNaN(avgYoungGCTime) && avgYoungGCTime > youngGenGcTimeThreshold);
+  }
+
+  private boolean prematurePromotionTooHigh(double avgGarbagePromoted) {
+    return (!Double.isNaN(avgGarbagePromoted)
+        && avgGarbagePromoted <= 1
+        && avgGarbagePromoted > GARBAGE_PROMOTION_THRESHOLD);
+  }
+
+  private double getFollowerCheckTimeoutMs() {
+    return FOLLOWER_CHECK_TIMEOUT_MS;
+  }
+
+  private ResourceFlowUnit<HotResourceSummary> compute(double avgPromotionRate,
+                                                       double avgYoungGCTime,
+                                                       double avgGarbagePromoted,
+                                                       double avgFullGCTime) {
+    ResourceContext context = new ResourceContext(State.UNHEALTHY);
+    HotResourceSummary summary = null;
+    boolean unhealthy = true;
+
+    // Check if the RCA is unhealthy
```

Comment:
should this be an if-else-if? or should we let the same node indicate multiple issues - they can be a signal of urgency to upstream consumers

Reply:
This is a good idea, but I'd rather implement this as a future iteration. For now, given the current aggregation strategy, a simple issue count is enough.